packages/cfd/src/Components/__tests__/cfd-poa.spec.js

```diff
@@ -0,0 +1,296 @@
+                date_of_birth: 984960000,
+                email: 'mock@gmail.com',
+                email_consent: 1,
+                feature_flag: {
+                    wallet: 0,
+                },
+                first_name: 'thisyahlen',
+                has_secret_answer: 1,
+                immutable_fields: ['residence'],
+                is_authenticated_payment_agent: 0,
+                last_name: 'lol',
+                non_pep_declaration: 1,
+                phone: '+790875616',
+                place_of_birth: null,
+                preferred_language: 'EN',
+                request_professional_status: 0,
+                residence: 'Singapore',
+                salutation: '',
+                tax_identification_number: null,
+                tax_residence: null,
+                user_hash: '823341c18bfccb391b6bb5d77ab7e6a83991f82669c1ba4e5b01dbd2fd71c7fe',
+            },
+            height: 'auto',
+            is_loading: false,
+            states_list: {
+                text: 'Central Singapore',
+                value: '01',
+            },
+            storeProofOfAddress: jest.fn(),
+            value: {
+                address_city: '',
+                address_line_1: '',
+                address_line_2: '',
+                address_postcode: '',
+                address_state: 'Default test state',
+                poa_status: 'verified',
+            },
+        };
+    });
+
+    it('should render the POA component', async () => {
+        render(<CFDPOA {...props} />);
+        await waitFor(() => {
+            expect(screen.getByText(/Address information/i)).toBeInTheDocument();
+        });
+    });
+
+    it('should render the POA component with the correct texts and buttons', async () => {
+        render(<CFDPOA {...props} />);
+        await waitFor(() => {
+            expect(screen.getByText(/Address information/i)).toBeInTheDocument();
+            expect(screen.getByText(/First line of address*/i)).toBeInTheDocument();
+            expect(screen.getByText(/Second line of address \(optional\)/i)).toBeInTheDocument();
+            expect(screen.getByText('Town/City*')).toBeInTheDocument();
+            expect(screen.getByText('State/Province')).toBeInTheDocument();
+            expect(screen.getByText('Postal/ZIP code')).toBeInTheDocument();
+            expect(screen.getByText(/FileUploaderContainer/i)).toBeInTheDocument();
+            expect(screen.getByRole('button', { name: /Previous/i })).toBeInTheDocument();
+            expect(screen.getByRole('button', { name: /Submit/i })).toBeInTheDocument();
+        });
+    });
+    it('should render the correct input for all the fields', async () => {
+        render(<CFDPOA {...props} onSubmit={() => true} />);
+
+        await waitFor(() => {
+            expect(screen.getByText(/Address information/i)).toBeInTheDocument();
+        });
+
+        const address_line_1_input = screen.getByPlaceholderText(address_line_1);
```

Comment:
isn't it better to put variable declarations at the beginning of the function?

Reply:
Will do